File: assignment7_1.py
import pandas as pd
from scipy.stats import ttest_ind


def calculate_statistics(female, male):
    # T-test
    t_statistic, p_value = ttest_ind(female, male)

    return t_statistic, p_value

def read_excel(file_path):
    # Read excel file
    df = pd.read_excel(file_path, sheet_name='Sheet2')

    # Data is in a column named 'close'
    female = df['Female']
    male = df['Male']

    return female, male

female, male = read_excel('Corn_bunting_bird.xlsx')
t_stat, p_value = calculate_statistics(female, male)

print("T-Statistic: ", t_stat)
print("P-Value: ", p_value)

File: assignment7_1.R
library("xlsx")
library("e1071")

# Read excel file
df <- read.xlsx("C:/Users/admin/OneDrive/Desktop/VIT/Sem 4/Foundataion of Data Science/R and Python Codes/Corn_bunting_bird.xlsx", sheetName = "Sheet2")


# Data is in a column named 'Data'
pre <- df$Female
post <- df$Male

# Perform t-test
t_test_result <- t.test(pre, post, var.equal = TRUE)

# Extract t-value and p-value
t_value <- t_test_result$statistic
p_value <- t_test_result$p.value

print_statistics <- function(t_value, p_value) {
    cat(paste("t-value:", t_value, "\n"))
    cat(paste("p-value:", p_value, "\n"))
}

# Call the function with the statistics
print_statistics(t_value, p_value)

